pkg/agents/team5/agent.go

```diff
@@ -0,0 +1,142 @@
+		lastMeal:          0,  //Stores value of the last amount of food taken
+		daysSinceLastMeal: 0,  //Count of how many days since last eating
+		hpAfterEating:     baseAgent.HealthInfo().MaxHP,
+		currentAimHP:      baseAgent.HealthInfo().MaxHP,
+		attemptFood:       0,
+		satisfaction:      0,                          // Scale of -3 to 3, with 3 being satisfied and unsatisfied
+		rememberAge:       -1,                         // To check if a day has passed by our age increasing
+		socialMemory:      make(map[uuid.UUID]Memory), // Memory of other agents, key is agent id
+		surroundingAgents: make(map[int]uuid.UUID),    // Map agent IDs of surrounding floors relative to current floor
+	}, nil
+}
+
+func (a *CustomAgent5) updateAimHP() {
+	a.currentAimHP = a.HealthInfo().MaxHP - ((10 - a.selfishness) * ((a.HealthInfo().MaxHP - a.HealthInfo().WeakLevel) / 10))
+}
+
+func (a *CustomAgent5) updateSelfishness() {
+	a.selfishness = 10 - a.calculateAverageFavour()
+}
+
+func (a *CustomAgent5) updateSatisfaction() {
+	if PercentageHP(a) >= 100 {
+		a.satisfaction = 3
+	}
+	if a.lastMeal == 0 && a.satisfaction > -3 {
+		a.satisfaction--
+	}
+	if PercentageHP(a) < 25 && a.satisfaction > -3 {
+		a.satisfaction--
+	}
+	if PercentageHP(a) > 75 && a.satisfaction < 3 {
+		a.satisfaction++
+	}
+}
+
+func (a *CustomAgent5) dayPassed() {
+	a.updateFavour()
+	a.updateSelfishness()
+	a.updateAimHP()
+	a.attemptFood = a.calculateAttemptFood()
+
+	// for id := range a.socialMemory {
+	// 	a.Log("Memory at end of day", infra.Fields{"favour": a.socialMemory[id].favour, "agentHP": a.socialMemory[id].agentHP, "foodTaken": a.socialMemory[id].foodTaken, "intent": a.socialMemory[id].intentionFood})
+	// }
+	// a.Log("Selfishness at end of day", infra.Fields{"selfishness": a.selfishness})
+	// a.Log("Aim HP at end of day", infra.Fields{"aim HP": a.currentAimHP})
+	// a.Log("Surrounding Agent Knowledge at end of day", infra.Fields{"agent map": a.surroundingAgents})
+
+	a.daysSinceLastMeal++
+	a.incrementDaysSinceLastSeen()
+	if a.rememberFloor != a.Floor() {
+		a.ResetSurroundingAgents()
+		a.rememberFloor = a.Floor()
+	}
+	a.messagingCounter = 0
+	a.rememberAge = a.Age()
+}
+
+func (a *CustomAgent5) calculateAttemptFood() food.FoodType {
+	if a.HP() < a.HealthInfo().WeakLevel {
+		// TODO: UPDATE THIS VALUE TO A PARAMETER
+		return food.FoodType(3)
+	}
+	foodAttempt := health.FoodRequired(a.HP(), a.currentAimHP, a.HealthInfo())
+	return food.FoodType(math.Min(a.HealthInfo().Tau*3, float64(foodAttempt)))
+}
+
+func (a *CustomAgent5) Run() {
+	a.Log("Reporting agent state of team 5 agent", infra.Fields{"health": a.HP(), "floor": a.Floor()})
+
+	//Check if a day has passed
+	if a.Age() > a.rememberAge {
+		a.dayPassed()
+	}
+
+	a.getMessages()
+	a.dailyMessages()
+
+	//When platform reaches our floor and we haven't tried to eat, then try to eat
+	if a.CurrPlatFood() != -1 && !a.HasEaten() {
+		lastMeal, err := a.TakeFood(a.attemptFood)
+		if err != nil {
```

Comment:
I see you logging errors to the default console logger here. These will never go to a logger file and we intended to use the console logger for the backend monitoring. Two propositions:
- Add an ID (e.g. "Simulation - {FILENAME}") next to it so we can identify it in the backend. You can even comment it out and I can deal with auto-id for these type of logs
- Use your agent logger, can add a log.Error to the base agent logging